src/modules/directives/gql.ts

```diff
@@ -1,20 +1,38 @@
 import {SchemaDirectiveVisitor, makeRemoteExecutableSchema, delegateToSchema, introspectSchema} from 'graphql-tools';
-import {GraphQLField} from 'graphql';
+import {GraphQLField, GraphQLSchema} from 'graphql';
 import {gql} from 'apollo-server-core';
 import {HttpLink} from 'apollo-link-http';
+import {setContext} from 'apollo-link-context';
 import fetch from 'node-fetch';
 import {injectParametersToObject} from '../param-injection';
+import {getAuthHeaders} from '../auth';
 
 export class GqlDirective extends SchemaDirectiveVisitor {
     visitFieldDefinition(field: GraphQLField<any, any>) {
         const {url, fieldName, arguments: gqlArgs, operationType: operationTypeParam} = this.args;
         const operationType = operationTypeParam?.toLowerCase() ?? 'query';
-        const link = new HttpLink({uri: url, fetch: fetch as any});
-        const remoteSchema = introspectSchema(link).then(schema => makeRemoteExecutableSchema({schema, link}));
+
+        const http = new HttpLink({uri: url, fetch: fetch as any});
+        let schemaInitialized = false;
+        let remoteSchema = {} as GraphQLSchema;
```

Comment:
Can we do something like `let remoteSchema : GraphQLSchema | null` and avoid using the `schemaInitialized` variable with this?